src/main/java/org/opensearch/security/tools/Hasher.java

```diff
@@ -82,11 +81,7 @@ public static void main(final String[] args) {
     }
 
     public static String hash(final char[] clearTextPassword) {
-        final byte[] salt = new byte[16];
-        new SecureRandom().nextBytes(salt);
-        final String hash = OpenBSDBCrypt.generate((Objects.requireNonNull(clearTextPassword)), salt, 12);
-        Arrays.fill(salt, (byte) 0);
-        Arrays.fill(clearTextPassword, '\0');
-        return hash;
+        PasswordHasher passwordHasher = new BCryptPasswordHasher();
```

Comment:
Is it possible to this tool to produce an output based on the configured password hashing strategy?

The purpose of the [`hash.sh` tool](https://github.com/opensearch-project/security/blob/main/tools/hash.sh#L35) is to help users configure security. Using the tool allows a user to pass a plaintext password and get a bcrypt hash of the password to put in `internal_users.yml` before starting a cluster for the first time.

Reply:
Yes, I have something in the works for this but I will push this as part of a future PR where we add extra BCrypt configuration. Initially I will be adding extra command line arguments for the users to be able to configure the version of BCrypt and the log rounds. Afterwards that will be expanded to support other hashing algorithms and the corresponding parameters for each alg.

Reply:
Got it. I re-read the PR description and see that this change only changes the BCrypt implementation from BC -> Password4j. I look forward to seeing the future PRs as well!